deps/rabbitmq_mqtt/src/rabbit_mqtt_qos0_queue.erl

```diff
@@ -126,6 +126,8 @@ delete(Q, _IfUnused, _IfEmpty, ActingUser) ->
     log_delete(QName, amqqueue:get_exclusive_owner(Q)),
     case rabbit_amqqueue:internal_delete(Q, ActingUser) of
         ok ->
+            Pid = amqqueue:get_pid(Q),
+            delegate:invoke_no_result([Pid], {gen_server, cast, [{queue_event, ?MODULE, {queue_down, QName}}]}),
```

Comment:
Why is `delegate` being used here? To me this seems an overkill. Just calling `gen_server:cast` directly is good enough.

Reply:
Can we emit an `eol` "end of life"  instead of `queue_down` event?
This way, we use a consistent naming with classic queues and quorum queues, and streams: They all emit an `eol` event if the queue is known to be deleted.

Reply:
> Same as above, in 0.9.1 the channel emits a queue_down event in this case

Where exactly is this happening? The fact that the channel handles 'DOWN' monitor messages is a relict from prior to the `rabbit_queue_type` API.
The more general direction after the `rabbit_queue_type` API got introduced is that an `eol` event/action queue action denotes a queue deletion.

Reply:
https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_channel.erl#L2750